bots/heroes/dr_main.lua

```diff
@@ -0,0 +1,372 @@
+behaviorLib.LateItems  = {"Item_Protect", "Item_Morph"}
+
+-- skillbuild table, 0=q, 1=w, 2=e, 3=r, 4=attri
+object.tSkills = {
+	0, 2, 1, 0,		-- Q E W Q	(4)
+	1, 3, 0, 0,		-- W R Q Q	(8)		Maxed Q here	(Level 1 ult)
+	1, 3, 1, 2,		-- W R W E  (12)	Maxed W here	(Level 2 ult)
+	2, 2, 2, 3,		-- E E E R  (16)					(Level 3 ult)
+	4, 4, 4, 4,		-- stats	(20)
+	4, 4, 4, 4,		-- stats	(24)
+	4				-- stats	(25)
+}
+
+-- bonus agression points if a skill/item is available for use
+object.nContrpUp = 20
+object.nOpposingUp = 10
+object.nSpeedUp = 40
+object.nHFlowerUp = 40
+object.nMorphUp = 45
+
+-- bonus agression points that are applied to the bot upon successfully using a skill/item
+object.nContrpUse = 30
+object.nOpposingUse = 15
+object.nSpeedUse = 35
+object.nHFlowerUse = 40
+object.nMorphUse = 40
+
+--thresholds of aggression the bot must reach to use these abilities
+object.nContrpThreshold = 25
+object.nOpposingThreshold = 10
+object.nSpeedThreshold = 40
+object.nHFlowerThreshold = 40
+object.nMorphThreshold = 40
+
+-- This is a static variable indicating that we have used our ultimate
+-- This is used so we can proc our E
+-- Note: Only used in aggression, see HarassHero
+object.bToggleUltimateFrenzy = false
+
+------------------------------
+--     skills               --
+------------------------------
+-- @param: none
+-- @return: none
+function object:SkillBuild()
+	local unitSelf = self.core.unitSelf
+	if skills.abilQ == nil then
+		skills.abilQ = unitSelf:GetAbility(0)
+		skills.abilW = unitSelf:GetAbility(1)
+		skills.abilR = unitSelf:GetAbility(3)
+		skills.abilAttributeBoost = unitSelf:GetAbility(4)
+	end
+
+	if unitSelf:GetAbilityPointsAvailable() <= 0 then
+		return
+	end
+
+	local nlev = unitSelf:GetLevel()
+	local nlevpts = unitSelf:GetAbilityPointsAvailable()
+	for i = nlev, nlev+nlevpts do
+		unitSelf:GetAbility( object.tSkills[i] ):LevelUp()
+	end
+end
+
+local function funcFindItemsOverride(botBrain)
+	local bUpdated = object.FindItemsOld(botBrain)
+
+	if core.itemMorph ~= nil and not core.itemMorph:IsValid() then
+		core.itemMorph = nil
+	end
+
+	if core.itemHFlower ~= nil and not core.itemHFlower:IsValid() then
+		core.itemHFlower = nil
+	end
+
+	if bUpdated then
+		if core.itemHFlower and core.itemMorph then
+			return
+		end
+
+		local inventory = core.unitSelf:GetInventory(true)
+		for slot = 1, 12, 1 do
+			local curItem = inventory[slot]
+			if curItem then
+				if core.itemMorph == nil and curItem:GetName() == "Item_Morph" then
+					core.itemMorph = core.WrapInTable(curItem)
+				elseif core.itemHFlower == nil and curItem:GetName() == "Item_Silence" then
+					core.itemHFlower = core.WrapInTable(curItem)
+				end
+			end
+		end
+	end
+end
+object.FindItemsOld = core.FindItems
+core.FindItems = funcFindItemsOverride
+
+-- @param: tGameVariables
+-- @return: none
+function object:onthinkOverride(tGameVariables)
+	self:onthinkOld(tGameVariables)
+
+	-- If we're low HP, run for our life!
+	local unitSelf = core.unitSelf
```

Comment:
This needs to be moved to behaviorLib.CustomRetreatExecute
Also you always move to same direction even when on hellbourne